session.go

```diff
@@ -73,9 +73,7 @@ func newSession(c *conn, channel uint16) *Session {
 		close:          make(chan struct{}),
 		done:           make(chan struct{}),
 	}
-}
 
```

Comment:
Moving 'init' into newSession makes total sense and makes me wonder why it was split apart at all.